DPPP/MSBDAReader.cc

```diff
@@ -209,12 +203,19 @@ bool MSBDAReader::process(const DPBuffer&) {
     }
 
     const auto ant12 = std::make_pair(ant1_col(nread_), ant2_col(nread_));
+    casacore::Complex* data_ptr = nullptr;
+    std::vector<casacore::Complex> data;
+    if (read_vis_data_) {
+      data = data_col.get(nread_).tovector();
+      data_ptr = data.data();
+    }
+    Cube<float> weights = weights_col.get(nread_);
+    Cube<double> uvw = uvw_col.get(nread_);
```

Comment:
Optional: You could perform the conversion from the casacore::Cube to std::vector here, instead of below(in the AddRow() call).